float32 X_dir
float32 Y_dir
float32 Z_dir
int32   Mode
bool    ModeChange
bool    ClawOpen
